deviation_analysis.py

```diff
@@ -0,0 +1,52 @@
+# script to run to analyze the deviation data. Histograms, boxplots, and metrics are generated for each currency analyzed.
+import pandas as pd
+import matplotlib.pyplot as plt
+import json
+
+THRESHOLD=0.1
+
+with open('dev_amplitude.json', 'r') as file:
+    data = json.load(file)
+
+dfs = {currency: pd.DataFrame(records) for currency, records in data.items()}
+
+for currency, df in dfs.items():
```

Comment:
As far as I understand, this dataframe is only filled if the price is not in the range 'ohlcvLow' < price < 'ohlcvHigh', see the conditions [here](https://github.com/pendulum-chain/pendulum-squids/pull/42/files#diff-08d6fa9cad6cb45bbcb2efc2f6f79e5c635d49266aa9b3087b27f8d1884b5506R61-R69). This makes sense to keep the dataframe low in size. 

However, this means that the 'mean deviation' that is calculated below is the 'mean deviation **of deviations**' and not of all prices since all the datapoints that are between 'low' and 'high' are not contained in the dataframe and not taken into account when calculating the average. I think that this is the reason why the results for the average deviations in your analysis are so high. It would be interesting to see how large the average deviation is when taking all prices into account not just the outliers.

Reply:
Exactly, It means the mean deviation of deviations just like the standard deviation is also of deviations, I forgot to mention that in the print line. It is not a measure of the overall precision itself. It tells us that if the price deviates, it will deviate by that much, but not necessarily a measure of how many times it deviates. Sadly the deviation events count is already very high in comparison with all the blocks analyzed.

To get the average deviation accounting for all blocks, we can in the python analysis script add all the entries for the remaining blocks assuming they deviation is 0 and recalculate the mean, which is a conservative but reasonable estimate given that the price was in between the high,low boundary.

I will modify so we get an account of how many blocks exactly were analyzed. This is possible to do right now from the difference in timestamp and assuming 12 seconds per block, but it may be better to have the number explicit.